drivers/eeprom/eeprom_simulator.c

```diff
@@ -193,17 +212,99 @@ static const struct eeprom_sim_config eeprom_sim_config_0 = {
 	.readonly = DT_INST_0_ZEPHYR_SIM_EEPROM_READ_ONLY,
 };
 
+#ifdef CONFIG_ARCH_POSIX
+
+static int eeprom_mock_init(struct device *dev)
+{
+	if (eeprom_file_path == NULL) {
+		eeprom_file_path = default_eeprom_file_path;
+	}
+
+	eeprom_fd = open(eeprom_file_path, O_RDWR | O_CREAT, (mode_t)0600);
+	if (eeprom_fd == -1) {
+		posix_print_warning("Failed to open eeprom device file "
+				    "%s: %s\n",
+				    eeprom_file_path, strerror(errno));
+		return -EIO;
+	}
+
+	if (ftruncate(eeprom_fd, DT_INST_0_ZEPHYR_SIM_EEPROM_SIZE) == -1) {
+		posix_print_warning("Failed to resize eeprom device file "
+				    "%s: %s\n",
+				    eeprom_file_path, strerror(errno));
+		return -EIO;
+	}
+
+	mock_eeprom = mmap(NULL, DT_INST_0_ZEPHYR_SIM_EEPROM_SIZE,
+			  PROT_WRITE | PROT_READ, MAP_SHARED, eeprom_fd, 0);
+	if (mock_eeprom == MAP_FAILED) {
+		posix_print_warning("Failed to mmap eeprom device file "
+				    "%s: %s\n",
+				    eeprom_file_path, strerror(errno));
+		return -EIO;
+	}
+
+	return 0;
+}
+
+#else
+
+static int eeprom_mock_init(struct device *dev)
+{
+	memset(mock_eeprom, 0xFF, ARRAY_SIZE(mock_eeprom));
+	return 0;
+}
+
+#endif /* CONFIG_ARCH_POSIX */
+
 static int eeprom_sim_init(struct device *dev)
 {
 	SYNC_INIT();
 	STATS_INIT_AND_REG(eeprom_sim_stats, STATS_SIZE_32, "eeprom_sim_stats");
 	STATS_INIT_AND_REG(eeprom_sim_thresholds, STATS_SIZE_32,
 			   "eeprom_sim_thresholds");
-	memset(mock_eeprom, 0xFF, ARRAY_SIZE(mock_eeprom));
+	eeprom_mock_init(dev);
```

Comment:
Better return the value of the function, just in case the posix arch version of`eeprom_mock_init ` returns an error code.